R/get_-functions.R

```diff
@@ -16,7 +16,9 @@
 get_forecast_type <- function(data) {
   assert_data_frame(data)
   assert(check_columns_present(data, c("observed", "predicted")))
-  if (test_forecast_type_is_binary(data)) {
```

Comment:
So this is a remaining pain point for the package right in terms of adding a new class. Do we have a specific issue for this? Seems tricky to resolve unless we can just rely on s3 dispatch?

We are using this as a check_forecast_type mostly right?

Reply:
I think we can simplify `get_forecast_type()` or even get rid of it entirely. Now that we have `as_forecast_<type>` I don't think we ever need to check for a forecast type without knowing the class of the object we're checking.

Reply:
Issue: https://github.com/epiforecasts/scoringutils/issues/887